### PR TITLE
adding the telemetry file for test using python code

### PR DESCRIPTION
**Pull request by ahmedDaoudi-u**
In the next commits, I will use the GraphQL that we have talked about before.

### CODE REVIEW

File: telemetry.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from flask import Flask, request
+from influxdb import InfluxDBClient
+
+app = Flask(__name__)
+
+client = InfluxDBClient(host='localhost', port=8086)
```
Comment (tot-ra) [inline telemetry.py:6]: I think in a dockerized env this will need to used service name of the influx container,
maybe we need some config to be extracted

Comment (ahmedDaoudi-u) [reply telemetry.py:6]: I see @tot-ra I will look into it

**Comment by ahmedDaoudi-u** [comment]
In this last commit I have added the docker files ,but I accounted an issue when running the docker-compose build.
image

**Comment by tot-ra** [comment]
@ahmedDaoudi-u it kinda worked for me (I did have some port conflict initially though)
Screenshot 2024-05-25 at 23 42 05

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM python:3.9-slim
```
Comment (tot-ra) [inline Dockerfile:1]: I think there are now newer python versions out, maybe could be upgraded

But as I said before, in general I very much doubt that python will be able to handle high volume traffic. Typically go or rust is better suited for this role.

But I guess we could have this as initial version to have at least some functionality today and we could rewrite it later

File: telemetry.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+influxdb_host = os.environ.get('INFLUXDB_HOST', 'localhost')
+influxdb_port = int(os.environ.get('INFLUXDB_PORT', 8086))
+influxdb_database = os.environ.get('INFLUXDB_DATABASE', 'beehive_data')
+
+client = InfluxDBClient(host=influxdb_host, port=influxdb_port)
+client.switch_database(influxdb_database)
+
+# Define the API endpoint to receive telemetry data
+@app.route('/telemetry', methods=['POST'])
```
Comment (tot-ra) [inline telemetry.py:15]: so as I was saying, we will need to use graphql federation for https://github.com/Gratheon/graphql-router to route traffic to this service, because of the efficiency and also because graphql-router does authentication of users

I don't know what graphql server supports graphql federation
plain google search points to
https://atechnologistspov.com/graphql-federation-in-python-46eaf8a4b185
https://github.com/graphql-python/graphene-federation
https://github.com/shaungehring/python-graphql-federation-sample

So basically instead of flask exposing a REST API, some server would need to expose POST /graphql endpoint that would do queries and mutations

And also you would need to make request to schema-registry to register your schema, an example is done here in JS - https://github.com/Gratheon/image-splitter/blob/main/src/graphql/schema-registry.ts

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+version: '3.8'
```
Comment (tot-ra) [inline docker-compose.yml:1]: on local machine I also have
```
    networks:
      - gratheon
networks:
  gratheon:
```
for example here in
https://github.com/Gratheon/image-splitter/blob/main/docker-compose.dev.yml#L23

This allows docker containers to work within a same network so that you could make requests to schema-registry by its name
Screenshot 2024-05-25 at 23 59 18

**Comment by tot-ra** [comment]
I also suggest to follow naming convention of `docker-compose.dev.yml` and `Dockerfile.dev` for development cases and have `.prod` suffix for production images. They tend to be similar, but this allows to customize a bit different behaviour in different environments

**Comment by ahmedDaoudi-u** [comment]
Great to read that , I will take into consideration next time , I will look into the GrapghQL as this was just for testing , I would change the code from Flask in python to rust.thanks @tot-ra
